docs/spec/tuples.rst

```diff
@@ -0,0 +1,154 @@
+``int``, and a ``str``.
+
+The empty tuple can be annotated as ``tuple[()]``.
+
+Arbitrary-length homogeneous tuples can be expressed using one type and
+ellipsis, for example ``tuple[int, ...]``. This type is equivalent to a union
+of tuples containing zero or more ``int`` elements (``tuple[()] |
+tuple[int] | tuple[int, int] | tuple[int, int, int] | ...``).
+Arbitrary-length homogeneous tuples are sometimes referred to as "unbounded
+tuples". Both of these terms appear within the typing spec, and they refer to
+the same concept.
+
+The type ``tuple[Any, ...]`` is special in that it is bidirectionally
+compatible with any tuple of any length. This is useful for gradual typing.
+The type ``tuple`` (with no type arguments provided) is equivalent to
+``tuple[Any, ...]``.
+
+Arbitrary-length tuples have exactly two type arguments -- the type and
+an ellipsis. Any other tuple form that uses an ellipsis is invalid::
+    
+    t1: tuple[int, ...]  # OK
+    t2: tuple[int, int, ...]  # Invalid
+    t3: tuple[...] # Invalid
+    t4: tuple[..., int] # Invalid
+    t5: tuple[int, ..., int] # Invalid
+    t6: tuple[*tuple[str], ...] # Invalid
+    t7: tuple[*tuple[str, ...], ...] # Invalid
+
+
+Unpacked Tuple Form
+-------------------
+
+An unpacked form of ``tuple`` (using an unpack operator ``*``) can be used
+within a tuple type argument list. For example, ``tuple[int, *tuple[str]]``
+is equivalent to ``tuple[int, str]``. Unpacking an unbounded tuple preserves
+the unbounded tuple as it is. That is, ``*tuple[int, ...]`` remains
+``*tuple[int, ...]``; there's no simpler form. This enables us to specify
+types such as ``tuple[int, *tuple[str, ...], str]`` -- a tuple type where the
+first element is guaranteed to be of type ``int``, the last element is
+guaranteed to be of type ``str``, and the elements in the middle are zero or
+more elements of type ``str``. The type ``tuple[*tuple[int, ...]]`` is
+equivalent to ``tuple[int, ...]``.
+
+If an unpacked ``*tuple[Any, ...]`` is embedded within another tuple, that
+portion of the tuple is bidirectionally type compatible with any tuple of
+any length.
+
+Only one unbounded tuple can be used with another tuple::
+
+    t1: tuple[*tuple[str], *tuple[str]]  # OK
+    t2: tuple[*tuple[str, *tuple[str, ...]]]  # OK
+    t3: tuple[*tuple[str, ...], *tuple[int, ...]]  # Type error
+    t4: tuple[*tuple[str, *tuple[str, ...]], *tuple[int, ...]]  # Type error
+
+An unpacked TypeVarTuple counts as an unbounded tuple in the context of this rule::
+
+    def func[*Ts](t: tuple[*Ts]):
+        t5: tuple[*tuple[str], *Ts]  # OK
+        t6: tuple[*tuple[str, ...], *Ts]  # Type error
+
+The ``*`` syntax requires Python 3.11 or newer. For older versions of Python,
+the ``typing.Unpack`` special form can be used:
+``tuple[int, Unpack[tuple[str, ...]], int]``.
+
+Unpacked tuples can also be used for ``*args`` parameters in a function
+signature. For example, ``def f(*args: *tuple[int, str]): ...``. Unpacked tuples
```

Comment:
Incomplete sentence, maybe add that this is equivalent to `def f(arg1: int, arg2: str, /)`.